rubix/galaxy/input_handler/illustris.py

```diff
@@ -54,9 +54,75 @@ def get_units(self):
         return self.UNITS
 
     def _check_fields(self, f):
-        for fields in self.ILLUSTRIS_DATA:
-            if fields not in f:
-                raise ValueError(f"Field {fields} not found in the file")
+        self._logger.debug("Checking if the fields are present in the file...")
+        self._logger.debug(f"Keys in the file: {f.keys()}")
+        self._logger.debug(f"Expected fields: {self.ILLUSTRIS_DATA}")
+
+        present_fields = set(f.keys())
+        expected_fields = set(self.ILLUSTRIS_DATA)
+
+        matching_fields = present_fields.intersection(expected_fields)
+        extra_fields = present_fields - expected_fields
+
+        if not matching_fields:
+            raise ValueError(
+                f"No expected fields found in the file. Expected at least one of: {self.ILLUSTRIS_DATA}"
+            )
+
+        for field in extra_fields:
+            if field.startswith("PartType"):
+                raise NotImplementedError(
+                    f"Unsupported particle type found in the file: {field}"
+                )
+            else:
+                raise ValueError(f"Unexpected fields found in the file: {extra_fields}")
+
+        self._logger.debug(f"Matching fields: {matching_fields}")
+
+    """
+    def _check_particle_data(self, particle_data, units):
```

Comment:
why is this commented out?

Reply:
Not sure, why it was commented out, now I included the part again